File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        (x1, y1), (x2, y2) = coordinates[:2]
        return all((x-x1) * (y2-y1) == (y-y1) * (x2-x1) for x, y in coordinates)

File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.js
/**
 * @param {number[][]} coordinates
 * @return {boolean}
 */
var checkStraightLine = function(coordinates) {
  const [[x1, y1], [x2, y2]] = coordinates.slice(0, 2);
  
  for (let i = 2; i < coordinates.length; i++) {
    const [x, y] = coordinates[i];
    if ((x2 - x1) * (y - y1) !== (y2 - y1) * (x - x1)) return false;
  }
  
  return true;
};